port/cpl_vsi_virtual.h

```diff
@@ -58,7 +58,7 @@
 /************************************************************************/
 
 /** Virtual file handle */
-class CPL_DLL VSIVirtualHandle
+struct CPL_DLL VSIVirtualHandle
```

Comment:
> Why this change ?

given that cpl_vsi.h has now ``typedef struct VSIVirtualHandle VSILFILE``, we need to be consistent. gcc would happily mix struct and class, but with clang, given test2.cpp with:
```
class VSIVirtualHandle { public: int i; };
typedef struct VSIVirtualHandle VSILFILE;
```
```
$ clang++ -c -Wall test2.cpp
test2.cpp:2:9: warning: struct 'VSIVirtualHandle' was previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
typedef struct VSIVirtualHandle VSILFILE;
        ^
test2.cpp:1:7: note: previous use is here
class VSIVirtualHandle { public: int i; };
      ^
test2.cpp:2:9: note: did you mean class here?
typedef struct VSIVirtualHandle VSILFILE;
        ^~~~~~
        class
1 warning generated.
```
And we can't  do ``typedef class VSIVirtualHandle VSILFILE`` since cpl_vsi.h needs to use C only features